drivers/ethernet/eth_stm32_hal.c

```diff
@@ -382,14 +493,33 @@ static enum ethernet_hw_caps eth_stm32_hal_get_capabilities(struct device *dev)
 {
 	ARG_UNUSED(dev);
 
-	return ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T;
+	return ETHERNET_LINK_10BASE_T | ETHERNET_LINK_100BASE_T 
+#if defined(CONFIG_PTP_CLOCK_STM32)
+	        |ETHERNET_PTP;
```

Comment:
Place space after | character

If you write the #ifdef like this, then you do not need the #else 
```
#if defined(CONFIG_PTP_CLOCK_STM32)
           | ETHERNET_PTP
#endif 
           ;
```